### PR TITLE
Issue 12544  Sum(cos(1/n) - 1, (n, 1, oo)).is_convergent() issue

### LINKED ISSUE #12544: Sum(cos(1/n) - 1, (n, 1, oo)).is_convergent() returns false
The following code

```
n = Symbol('n')
expr = cos(1/n) - 1
print(Sum(expr, (n, 1, oo)).is_convergent())

```
returns `False`. This is not correct. Indeed this series has the same behavior of `Sum(1/n**2)`, which is convergent.

### PR DESCRIPTION
**Pull request by maurogaravello**
The series of `cos(1/n) - 1` is convergent, but the
`is_convergent()` method return false.
To fix this issue, a function for finding the asymptotic behavior
is implemented.

The changes are in `summation.py`.
A test is added.

"Fixes #12544 "

### CODE REVIEW

File: sympy/concrete/summations.py [hunk 1]
```diff
@@ -1098,3 +1099,35 @@ def eval_sum_hyper(f, i_a_b):
...
         return Piecewise(res, (old_sum, True))
+
+def asymptotic(expr, symbol):
+    '''Tries to perform the asymptotic of expr when
+    symbols tends to infinity
+
+    return either the asymptotic expression or the original one
+    '''
+    from sympy import simplify
+    if type(expr) is Mul:
```
Comment (asmeurer) [inline sympy/concrete/summations.py]: Use `isinstance` to check types.

File: sympy/concrete/summations.py [hunk 2]
```diff
@@ -1098,3 +1099,35 @@ def eval_sum_hyper(f, i_a_b):
...
+        if lim != S(0): # the asymptotic is the value of the limit
+            return lim
+        expr2 = expr.subs(symbol, 1/symbol)
+        s = expr2.series(symbol, 0, 10) # use MacLaurin expansion
+        if type(s) is Add:
+            s = s.subs(symbol, 1/symbol)
+            return s.args[0]
+        else:
+            return expr
+    except:
```
Comment (asmeurer) [inline sympy/concrete/summations.py]: Don't use bare excepts. Catch only the exception you are expecting.

File: sympy/concrete/summations.py [hunk 3]
```diff
@@ -1098,3 +1099,35 @@ def eval_sum_hyper(f, i_a_b):
                     return S.NegativeInfinity
             return None
         return Piecewise(res, (old_sum, True))
+
+def asymptotic(expr, symbol):
+    '''Tries to perform the asymptotic of expr when
```
Comment (asmeurer) [inline sympy/concrete/summations.py]: asymptotic expansion?

File: sympy/concrete/summations.py [hunk 4]
```diff
@@ -6,7 +6,7 @@
 from sympy.core.relational import Eq
 from sympy.core.singleton import S
 from sympy.core.symbol import Dummy, Wild, Symbol
-from sympy.core.add import Add
+from sympy.core.add import Add, Mul
```
Comment (asmeurer) [inline sympy/concrete/summations.py]: Mul should come from sympy.core.mul.

**Review by asmeurer — CHANGES_REQUESTED** [review]

File: sympy/concrete/summations.py (hunk 3 above)
Comment (maurogaravello) [inline sympy/concrete/summations.py]: Not always is an expansion. Probably 'asymptotic bahavior' is better

File: sympy/concrete/summations.py (hunk 2 above)
Comment (maurogaravello) [inline sympy/concrete/summations.py]: I put `except Exception:`. I am not sure if it is ok. I tried with `except NotImplementedError:`, but some tests failed.

**Comment by maurogaravello** [comment]
In the case log(n/(n+1)) the method asymptotic does not work correctly. I will try to fix.

**Comment by maurogaravello** [comment]
@asmeurer What do you think with this PR?

File: sympy/concrete/summations.py (hunk 1 above)
Comment (maurogaravello) [inline sympy/concrete/summations.py]: Done

File: sympy/concrete/summations.py (hunk 4 above)
Comment (maurogaravello) [inline sympy/concrete/summations.py]: Done

**Pull request closed without merging**